xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-test/xwiki-platform-image-lightbox-test-pageobjects/src/main/java/org/xwiki/image/lightbox/test/po/Lightbox.java

```diff
@@ -0,0 +1,182 @@
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+
+    public boolean waitForSlide(int index)
+    {
+        try {
+            By slideSelector = By.cssSelector("#blueimp-gallery .slide-active");
+            getDriver().waitUntilCondition(attributeToBe(slideSelector, "data-index", String.valueOf(index)));
+            return getDriver().findElement(slideSelector).isDisplayed();
+        } catch (TimeoutException e) {
+            return false;
+        }
+    }
+
+    public WebElement getSlideElement()
+    {
+        By slideSelector = By.cssSelector("#blueimp-gallery .slide-active");
+        return getDriver().findElement(slideSelector);
+    }
+
+    public String getSlideIndex()
+    {
+        By slideSelector = By.cssSelector("#blueimp-gallery .slide-active");
+        return getDriver().findElement(slideSelector).getAttribute("data-index");
+    }
+
+    public String getCaption()
+    {
+        By captionSelector = By.cssSelector("#blueimp-gallery .caption");
+        return getDriver().findElement(captionSelector).getText();
+    }
+
+    public String getTitle()
+    {
+        By titleSelector = By.cssSelector("#blueimp-gallery .title");
+        return getDriver().findElement(titleSelector).getText();
+    }
+
+    public String getPublisher()
+    {
+        By publisherSelector = By.cssSelector("#blueimp-gallery .publisher");
+        return getDriver().findElement(publisherSelector).getText();
+    }
+
+    public String getDate()
+    {
+        By dateSelector = By.cssSelector("#blueimp-gallery .date");
+        return getDriver().findElement(dateSelector).getText().replace("on", "");
+    }
+
+    public WebElement getDownloadButton()
+    {
+        By downloadSelector = By.cssSelector("#blueimp-gallery .download");
+        return getDriver().findElement(downloadSelector);
+    }
+
+    public boolean isImageMissing()
+    {
+        try {
+            return getDriver().findElement(By.cssSelector("#blueimp-gallery .slide-error")).isDisplayed();
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+
+    public void close()
+    {
+        By escapeSelector = By.cssSelector("#blueimp-gallery .escape");
+        getDriver().findElement(escapeSelector).click();
+        getDriver().waitUntilElementDisappears(By.className("blueimp-gallery-display"));
+    }
+
+    public void next(int nextSlideIndex)
+    {
+        By nextSelector = By.cssSelector("#blueimp-gallery .next");
+        getDriver().findElement(nextSelector).click();
+
+        waitForSlide(nextSlideIndex);
+    }
+
+    public void previous(int nextSlideIndex)
+    {
+        By prevSelector = By.cssSelector("#blueimp-gallery .prev");
+        getDriver().findElement(prevSelector).click();
+
+        waitForSlide(nextSlideIndex);
+    }
+
+    public Optional<WebElement> toggleFullscreen()
```

Comment:
What could you do with the returned ``WebElement``? For me it makes sense to know the state, i.e. whether fullscreen is on or off after the method call, so a boolean is enough.